logstash-core/lib/logstash/instrument/metric_store.rb

```diff
@@ -51,12 +51,12 @@ def fetch_or_store(namespaces, key, default_value = nil)
       # BUT. If the value is not present in the `@fast_lookup` the value will be inserted and we assume that we don't
       # have it in the `@metric_store` for structured search so we add it there too.
 
-      value = @fast_lookup.get(namespaces.dup << key)
+      value = @fast_lookup.get((namespaces.dup << key).hash)
```

Comment:
@original-brownbear - Is this effectively stating that JRuby's Array.hash is broken ?

I am not sure I agree with the math of ```5%```,  that seems really high. If these hashes were persisted across 100k runs maybe.  I haven't done the math, but it also seems that to do the math would require the exact algorithm, not just the bit space. 

I am not concerned about chances of collision between a couple dozen values using a well established hashing algorithm.